fintorch/datasets/marketdata.py

```
@@ -0,0 +1,227 @@
+                + (pol.col("time_id") * pol.duration(seconds=1))
+            ).alias("ds")
+        )
+
+    def setup_directories(self) -> None:
+        try:
+            os.makedirs(self.root, exist_ok=True)
+            os.makedirs(os.path.join(self.root, "raw"), exist_ok=True)
+            os.makedirs(os.path.join(self.root, "processed"), exist_ok=True)
+        except OSError as e:
+            logging.error(f"Failed to create directories: {str(e)}")
+            raise RuntimeError(f"Failed to set up directories: {str(e)}")
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)